src/cleaner/cleaner_main.py

```diff
@@ -76,31 +76,20 @@ def sync(self):
             time.sleep(1)
 
 
-def get_worker(arg):
-    if arg == "docker_cache":
-        worker = Worker(clean_docker_cache.check_and_clean, 50, timeout=timedelta(minutes=10), cool_down_time=1800)
-    elif arg == "deleted_files":
-        worker = Worker(check_deleted_files.list_and_check_files, None, timeout=timedelta(minutes=10), cool_down_time=1800)
-    else:
-        raise ValueError("arguments %s is not supported.", arg)
-    return worker
-
-
-liveness_files = {
-    "docker_cache": "docker-cache-cleaner-healthy",
-    "deleted_files": "deleted-files-cleaner-healthy"
-}
+def get_worker(threshold):
```

Comment:
Doesn't plan support multiworker?

Reply:
agree. If you plan to support multiworker in the future. you can bring it back when you need it. the current implementation is a bit obscure.


---
In reply to: [252962426](https://github.com/Microsoft/pai/pull/2119#discussion_r252962426) [](ancestors = 252962426)

Reply:
Sure in current version, we can remove the worker wrap. Zhaoyu's original design is separate the cleaner into two parts: docker cleaner and file cleaner, and now the clean process is all based on docker. Later if we want to manage diskspace in different dimensions, we will bring it back.